src/js/stores/MesosSummaryStore.js

```diff
@@ -178,6 +178,42 @@ class MesosSummaryStore extends GetSetBaseStore {
     });
   }
 
+  getFrameworkTasksByName(serviceName = '') {
+    let frameworks = this.get('frameworks');
+
+    if (frameworks == null) {
+      return false;
+    }
+
+    return frameworks.filter((framework) => {
+      return framework.name === serviceName;
+    }).reduce((memo, framework) =>{
```

Comment:
This seems like a weird process to me. Aren't we ever going to find at most one?
And if we find multiple we will override it with the one found last? Also, if there are multiple why not start with `{TASK_ERROR: 0,...}` as the initial value?

Assuming we will only find one, why not do something like this:
```js
getFrameworkTasksByName(serviceName) {
  let frameworks = this.get('frameworks') || [];
  let framework = frameworks.find(function (framework) {
    return framework.name === serviceName;
  });
  if (!framework) {
    return null;
  }

  return Object.assign({TASK_ERROR: 0, ...}, framework);
}
```